src/components/Ingredients.js

```diff
@@ -8,21 +8,30 @@ class Ingredients extends React.Component {
     };
 
     this.componentDidMount = this.componentDidMount.bind(this);
+    this.componentWillUnmount = this.componentWillUnmount.bind(this);
+    this.abortController = new AbortController()
   }
 
   componentDidMount() {
     const api = "https://spoon-call.herokuapp.com/ingredients?id="+this.props.id
-    fetch(api)
+    fetch(api, {signal: this.abortController.signal})
```

Comment:
Does this fetch need to be wrapped in a `try` to then catch the error?

Reply:
That's not how I understand fetch. The error catching happens at the end of the fetch method.

Reply:
Ah. Cool cool. You are correct. Sorry - was thinking of something else - where you can have a general `try` and `catch` that you could wrap a number of promises over and just catch the one universal error.